x-pack/plugins/enterprise_search/server/utils/create_pipeline_definitions.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+export interface CreatedPipelines {
+  created: string[];
+}
+
+/**
+ * Used to create index-specific Ingest Pipelines to be used in conjunction with Enterprise Search
+ * ingestion mechanisms. Three pipelines are created:
+ * 1. `<indexName>@ml-inference` - empty by default. Any ML Inference processors that are enabled
+ *   for this index are intended to be encapsulated in this pipeline.
+ * 2. `<indexName>@custom` - empty by default. The customer is encouraged to edit this pipeline
+ *   manually in order to meet their business needs.
+ * 3. `<indexName>` - includes processors for binary content extraction, whitespace reduction, and nests
+ *   the above pipelines in itself via pipeline processors.
+ * @param indexName the index for which the pipelines should be created.
+ * @param esClient the Elasticsearch Client with which to create the pipelines.
+ */
+export const createIndexPipelineDefinitions = (
```

Comment:
@sphilipse this doesn't currently:
1) check that the index exists
2) check the license level

I think you'd said somewhere that that's fine, because we can do these in the UI, but let me know if we need to gate the backend logic.

Reply:
Just wanted to chime in here real quickly to say that I noticed the ML plugin has an [interesting solution to handling Kibana API endpoints that should be guarded by the license level](https://github.com/elastic/kibana/blob/fdbf1829cef3dc48ba73064edb8b9a2508eec5fd/x-pack/plugins/ml/server/lib/route_guard.ts#L73).

Reply:
I have a separate story to look into licensing and the best way to handle it. For now my plan is to definitely do it in the UI, and maybe add it to the API endpoints if that's easy enough (it should be, but I won't definitively know till I try). For now you can leave this ungated.